### DISCUSSION TITLE
Reading values of branches with dots in names

### DISCUSSION THREAD

**Discussion by lkildetoft** [scikit-hep/uproot5#313] [Q&A]
Hi! I'm having trouble reading leaves of branches with uproot, where the branch names contain dots.
For example in my ROOT file, I have the branch "EventInfoAux.", which contains a leaf "eventNumber". 
Normally I would access this with something like:
file["tree/EventInfoAux.eventNumber"]

However, this is not possible as it results in 
file["tree/EventInfoAux..eventNumber"]

which uproot doesn't understand. Is there an easy solution to this? 
Thanks!

**Comment by tamasgal** [comment]
Can you show more output from that file and also the error message? What is e.g. `file["tree"].keys()`?

**Comment by jpivarski** [comment]
Uproot doesn't do any special interpretation of the dots (`.`), though it does parse by slashes (`/`). ROOT inserts dots in branch names according to its C++ object-splitting rules and Uproot takes a hands-off approach to them.

You can check to see if this string is exactly what you get from [TTree.keys](https://uproot.readthedocs.io/en/latest/uproot.behaviors.TTree.TTree.html#keys), possibly with `recursive=False` to narrow in.

It's also possible for ROOT to make a file with non-unique branch names. Uproot will pick the _first_ that matches, at each level of hierarchy, if that's relevant. In this case, you need to select them by integer index. You can reverse-index them with [TBranch.index](https://uproot.readthedocs.io/en/latest/uproot.behaviors.TBranch.TBranch.html#index) (to find out which index to pass).
